src/WebJobs.Script/Binding/GeneralScriptBindingProvider.cs

```diff
@@ -0,0 +1,163 @@
+        {
+            string name = context.Type;
+            var attrType = this._tooling.GetAttributeTypeFromName(name);
+            if (attrType == null)
+            {
+                binding = null;
+                return false;
+            }
+
+            var attr = this._tooling.GetAttribute(attrType, context.Metadata);
+                        
+            binding = new GeneralScriptBinding(this._tooling, attr, context);
+            return true;
+        }
+
+        public override bool TryResolveAssembly(string assemblyName, out Assembly assembly)
+        {
+            return this._tooling.TryResolveAssembly(assemblyName, out assembly);
+        }
+
+
+        // Function.json specifies a type via optional DataType and Cardinality properties. 
+        // Read the properties and convert that into a System.Type. 
+        public static Type GetRequestedType(ScriptBindingContext context)
+        {
+            Type type = ParseDataType(context);
+
+
+            Cardinality cardinality;
+            if (!Enum.TryParse<Cardinality>(context.Cardinality, true, out cardinality))
+            {
+                cardinality = Cardinality.One; // default 
+            }
+
+            if (cardinality == Cardinality.Many)
+            {
+                // arrays are supported for both trigger input as well
+                // as output bindings
+                type = type.MakeArrayType();
+            }
+            return type;
+        }
+
+        // Parse the DataType field and return as a System.Type.
+        // Never return null. Use typeof(object) to refer to an unnkown. 
+        private static Type ParseDataType(ScriptBindingContext context)
+        {
+            DataType result;
+            if (Enum.TryParse<DataType>(context.DataType, true, out result))
+            {
+                switch (result)
+                {
+                    case DataType.Binary:
+                        return typeof(byte[]);
+
+                    case DataType.Stream:
+                        return typeof(Stream);
+
+                    case DataType.String:
+                        return typeof(string);
+                }
+            }
+
+            return typeof(object);
+        }
+
+        private class GeneralScriptBinding : ScriptBinding, IResultProcessingBinding
+        {
+            private readonly Attribute _attribute;
+            private readonly IJobHostMetadataProvider _tooling;
+
+            private Type _defaultType;
+
+            private MethodInfo _applyReturn; // Action<object,object>
+
+            public GeneralScriptBinding(IJobHostMetadataProvider tooling, Attribute attribute, ScriptBindingContext context)
+                : base(context)
+            {
+                _tooling = tooling;
+                _attribute = attribute;
+
+                _applyReturn = attribute.GetType().GetMethod("ApplyReturn", BindingFlags.Static | BindingFlags.Public | BindingFlags.NonPublic);              
```

Comment:
tactical support for custom return values. Need a better solution, tracked with https://github.com/Azure/azure-webjobs-sdk/issues/1104